AuthenticatorShared/Core/Vault/Repositories/AuthenticatorItemRepository.swift

```diff
@@ -176,14 +197,35 @@ extension DefaultAuthenticatorItemRepository: AuthenticatorItemRepository {
         return try await items.asyncMap { item in
             try await cryptographyService.decrypt(item)
         }
-        .compactMap { $0 }
     }
 
     func fetchAuthenticatorItem(withId id: String) async throws -> AuthenticatorItemView? {
         guard let item = try await authenticatorItemService.fetchAuthenticatorItem(withId: id) else { return nil }
         return try? await cryptographyService.decrypt(item)
     }
 
+    func refreshTotpCodes(on items: [ItemListItem]) async throws -> [ItemListItem] {
+        try await items.asyncMap { item in
+            guard case let .totp(model) = item.itemType,
+                  let key = model.itemView.totpKey,
+                  let keyModel = TOTPKeyModel(authenticatorKey: key)
+            else {
+                errorReporter.log(error: TOTPServiceError
+                    .unableToGenerateCode("Unable to refresh TOTP code for list view item: \(item.id)"))
+                return item
```

Comment:
🤔 I know this scenario is highly unlikely to happen, but shouldn't we return `nil` here and `.compactMap` the result of the `asyncMap`? Not quite understanding why it would return the `item` if it can't create the `TOTPKeyModel`.

Reply:
I figured that it would be better to keep the item in the list with a stale code rather than have it suddenly disappear, especially because if a new type of item is added at some point, this would still return the item as-is.